vpc/allocate/ip_pool_manager.go

```diff
@@ -42,38 +39,31 @@ func (mgr *IPPoolManager) lockConfiguration(parentCtx *context.VPCContext) (*fsl
 	return parentCtx.FSLocker.ExclusiveLock(path, &timeout)
 }
 
-func (mgr *IPPoolManager) assignMoreIPs(ctx *context.VPCContext, batchSize int, ipRefreshTimeout time.Duration) error {
-	if len(mgr.networkInterface.GetIPv4Addresses()) >= vpc.GetMaxIPv4Addresses(ctx.InstanceType) {
+func assignMoreIPs(ctx *context.VPCContext, batchSize int, ipRefreshTimeout time.Duration, networkInterface ec2wrapper.NetworkInterface, ipAssignmentFunction ec2wrapper.IPAssignmentFunction, ipFetchFunction ec2wrapper.IPFetchFunction) error {
+	if len(ipFetchFunction()) >= vpc.GetMaxIPAddresses(ctx.InstanceType) {
```

Comment:
Might be worth moving the various `len(ipFetchFunction())` calls to a variable for readability.